src/network/serverpackethandler.cpp

```diff
@@ -1358,7 +1358,8 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 		Check that target is reasonably close
 		(only when digging or placing things)
 	*/
-	if (action == 0 || action == 2 || action == 3) {
+	if ((!isSingleplayer() && !g_settings->getBool("disable_anticheat")) &&
+			(action == 0 || action == 2 || action == 3)) {
```

Comment:
iirc, the server can't change from single-player mode to multiplayer mode during run time.
and since it isn't a global cache (as in accessible in content_sao too), the output of (!isSingleplayer() && !g_settings->getBool("disable_anticheat")) can be cache instead and it would be faster to check a static bool